src/index.js

```diff
@@ -1,9 +1,36 @@
-var core = module.exports = require('./core');
-
-// plugins:
-core.extras         = require('./extras');
-core.filters        = require('./filters');
-core.interaction    = require('./interaction');
-core.loaders        = require('./loaders');
-core.spine          = require('./spine');
-core.text           = require('./text');
+var core = require('./core'),
+	math = core.math,
+	// plugins:
+	extras         = require('./extras'),
+	filters        = require('./filters'),
+	interaction    = require('./interaction'),
+	loaders        = require('./loaders'),
+	spine          = require('./spine'),
+	text           = require('./text'),
+    deprecation    = require('./deprecation');
+
+var modules = [];
+
+modules.push(math);
+modules.push(extras);
+modules.push(filters);
+modules.push(interaction);
+modules.push(loaders);
+modules.push(spine);
+modules.push(text);
+
+for (var i = 0; i < modules.length; i++) {
+
+	// Adds the module to the core, so you can do
+	// PIXI.Text instead of PIXI.text.Text
+	bundleUp(core,modules[i]);
+};
+
+function bundleUp(object1,object2){
+
+    for(var keys in object2){
+        object1[keys] = object2[keys];
+    }
+}
+
+module.exports = core;
```

Comment:
Keen to get your thoughts on this one @englercj . I agree that the namespacing is nice for modular usage. Very keen to let people build the custom versions of pixi too (omitting things they don't need)

I also think that doing the above will make the transition from v2 - v3 super easy when it comes to general usage. Its a small thing but ```PIXI.Text``` is less effort than ```PIXI.text.Text``` with the added bonus that people who use v2 will already know how to use v3.

Maybe we make it optional? What do ya think the best way to achieve both these things are?